### PR TITLE
code refactoring

### PR DESCRIPTION
**Pull request by amjadhifthikar**

### CODE REVIEW

File: jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java [hunk 1]
```diff
@@ -122,12 +124,14 @@ public synchronized boolean send(CarbonMessage carbonMessage, CarbonCallback car
...
-                jmsConnectionFactory.closeConnection(connection);
-            } catch (JMSConnectorException e) {
-                throw new ClientConnectorException(e.getMessage(), e);
+            if (jmsConnectionFactory != null) {
+                try {
+                    jmsConnectionFactory.closeMessageProducer(messageProducer);
+                    jmsConnectionFactory.closeSession(session);
+                    jmsConnectionFactory.closeConnection(connection);
+                } catch (JMSConnectorException e) {
+                    logger.error("Exception occured when closing connection " + e.getMessage());
```
Comment (Kishanthan) [inline jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java]: "e" is lost here. You have to include the "e" to get the stack trace.

File: jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java [hunk 2]
```diff
@@ -493,7 +490,6 @@ public MessageConsumer createMessageConsumer(Session session, Destination destin
      * @param session     JMS Session to create the producer
      * @param destination JMS destination which the producer should publish to
      * @return MessageProducer, who publish messages to particular destination with the given session
-     * @throws JMSConnectorException JMS Connector Exception
```
Comment (Kishanthan) [inline jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java:496]: Method signature has "throws JMSConnectorException", then why do we remove it form the doc?

**Review by Kishanthan — CHANGES_REQUESTED** [review]

**Review by Kishanthan — APPROVED** [review]

File: jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java (hunk 1 above)
Comment (amjadhifthikar) [reply jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java]: This is fixed.

File: jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java (hunk 2 above)
Comment (amjadhifthikar) [reply jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java:496]: This is fixed. The exception is added.

**Pull request merged**